docs/advanced-features/compiler.md

```diff
@@ -0,0 +1,131 @@
+---
+description: Learn about the Next.js Compiler, written in Rust, which transforms and minifies your Next.js application.
+---
+
+# Next.js Compiler
+
+<details open>
+  <summary><b>Version History</b></summary>
+
+| Version   | Changes                                                         |
+| --------- | --------------------------------------------------------------- |
+| `v12.0.0` | Next.js Compiler [introduced](https://nextjs.org/blog/next-12). |
+
+</details>
+
+The Next.js Compiler, written in Rust using [SWC](http://swc.rs/), allows Next.js to transform and minify your JavaScript code for production. This replaces Babel for individual files and Terser for minifying output bundles.
+
+Compilation using the Next.js Compiler is 17x faster than Babel and enabled by default using Next.js 12. If you have an existing Babel configuration or are using [unsupported features](#unsupported-features), your application will opt-out of the Next.js Compiler and continue using Babel.
```

Comment:
```suggestion
Compilation using the Next.js Compiler is 17x faster than Babel and enabled by default since Next.js version 12. If you have an existing Babel configuration or are using [unsupported features](#unsupported-features), your application will opt-out of the Next.js Compiler and continue using Babel.
```